seqr/views/apis/auth_api.py

```diff
@@ -31,10 +39,102 @@ def login_view(request):
     if not u:
         return create_json_response({}, status=401, reason='Invalid credentials')
 
+    request.session['anvil'] = None
+
     login(request, u)
 
     return create_json_response({'success': True})
 
+@csrf_exempt
+def login_google(request):
+  # Create flow instance to manage the OAuth 2.0 Authorization Grant Flow steps.
+  flow = google_auth_oauthlib.flow.Flow.from_client_secrets_file(
+      CLIENT_SECRETS_FILE, scopes=scopes)
+
+  # The URI created here must exactly match one of the authorized redirect URIs
+  # for the OAuth 2.0 client, which you configured in the API Console. If this
+  # value doesn't match an authorized URI, you will get a 'redirect_uri_mismatch'
+  # error.
+  flow.redirect_uri = 'http://localhost:3000/oauth2callback'
+
+  authorization_url, state = flow.authorization_url(
+      # Enable offline access so that you can refresh an access token without
+      # re-prompting the user for permission. Recommended for web server apps.
+      access_type='offline',
+      # Enable incremental authorization. Recommended as a best practice.
+      include_granted_scopes='true')
+
+  # Store the state so the callback can verify the auth server response.
+  request.session['state'] = state
```

Comment:
The example from Google guides uses a different web server. In the sample code, it stores the state in the sessions managed by the web server so that it can get the right state when it receives the OAuth2 callback from the authentication server.